Program.fs

```diff
@@ -54,21 +54,18 @@ module MainModule =
     let apiDomain = Env.varDefault "ARTIST_TALLY_API_DOMAIN" "localhost:3000"
     let isLive = (Env.varDefault "ARTIST_TALLY_ENV" "development") = "production"
 
-    let buildMessage<'bodyType> method url authHeader (body: 'bodyType option) = 
-        let uri = Uri url
-        let message = new Http.HttpRequestMessage ()
-        message.Method <- method
-        message.RequestUri <- uri
+    let buildMessage method url authHeader body = 
+        let message = new Http.HttpRequestMessage (Method = method, RequestUri = Uri url)
 
-        if body.IsSome then
-            let serialized = JsonConvert.SerializeObject body.Value
+        body
+        |> Option.iter (fun body ->
+            let serialized = JsonConvert.SerializeObject body
             let content = new Http.StringContent (serialized, Text.Encoding.UTF8, "application/json")
 
-            message.Content <- content
+            message.Content <- content)
 
-        match authHeader with
-        | Some header -> message.Headers.Authorization <- header
-        | None -> ignore ()    
+        authHeader
```

Comment:
Pattern match on options when you have two different (larger) things happen in the two cases. Otherwise, `Option.iter/`Option.map` make it clearer what's going on.